java/src/main/kotlin/io/spine/protodata/java/MethodCall.kt

```diff
@@ -0,0 +1,129 @@
+ * @param name The name of the method.
+ * @param arguments The list of the arguments passed to the method.
+ * @param generics The list of the type arguments passed to the method.
+ */
+public open class MethodCall<T> @JvmOverloads constructor(
+    private val scope: JavaElement,
+    name: String,
+    arguments: List<Expression<*>> = listOf(),
+    generics: List<JavaTypeName> = listOf()
+) : Expression<T>(
+    "${scope.toCode()}.${generics.genericTypes()}$name(${arguments.formatParams()})"
+) {
+
+    /**
+     * An expression of a Java method call.
+     *
+     * Can be a static or an instance method. In the case of the former, the scope is a class name.
+     * In the case of the latter — an object reference.
+     *
+     * @param T The method returned type.
+     *
+     * @param scope The scope of the method invocation: an instance receiving the method call, or
+     *   the name of the class declaring a static method.
+     * @param name The name of the method.
+     * @param argument The argument passed to the method.
+     * @param generics The list of the type arguments passed to the method.
+     */
+    public constructor(
+        scope: JavaElement,
+        name: String,
+        argument: Expression<*>,
+        generics: List<JavaTypeName> = listOf()
+    ) : this(scope, name, listOf(argument), generics)
+
+    /**
+     * Constructs an expression of calling another method on the result of this method call.
+     */
+    @JvmOverloads
+    public fun <R> chain(method: String, arguments: List<Expression<*>> = listOf()): MethodCall<R> =
+        MethodCall(this, method, arguments)
+
+    /**
+     * Constructs an expression of calling another method on the result of this method call.
+     */
+    public fun <R> chain(method: String, vararg argument: Expression<*>): MethodCall<R> =
+        MethodCall(this, method, argument.toList())
+
+    /**
+     * Constructs an expression chaining a call of the `build()` method.
+     */
+    public fun <R> chainBuild(): MethodCall<R> = chain("build")
+
+    /**
+     * Constructs an expression chaining a setter call.
+     */
+    public fun chainSet(field: String, value: Expression<*>): MethodCall<Message.Builder> =
+        fieldAccess(field).setter(value)
+
+    /**
+     * Constructs an expression chaining a call of an `addField(...)` method.
+     */
+    public fun chainAdd(field: String, value: Expression<*>): MethodCall<Message.Builder> =
+        fieldAccess(field).add(value)
+
+    /**
+     * Constructs an expression chaining a call of an `addAllField(...)` method.
+     */
+    public fun chainAddAll(field: String, value: Expression<*>): MethodCall<Message.Builder> =
+        fieldAccess(field).addAll(value)
+
+    private fun fieldAccess(fieldName: String) = FieldAccess(Expression(toCode()), fieldName)
+}
+
+/**
+ * The assumed reference to `this` when invoking a [MethodCall] within an instance.
+ */
+public class InstanceScope<T> : Expression<T>("")
```

Comment:
This would not work with the current implementation of `MethodCall` because it still has the dot notation in the string template (per my previous comments). 

I'm not suggesting to change the implementation of `MethodCall` now. Let's remove `InstanceScope` to avoid the misuse. We may return to revisiting `MethodCall` implementation later and then reintroduce the `InstanceScope` class.